kong/plugins/rate-limiting/policies/init.lua

```diff
@@ -252,6 +366,26 @@ return {
         kong.log.err("failed to set Redis keepalive: ", err)
       end
 
+      if conf.sync_rate > 0 then
+        if cur_usage[cache_key] then
+          --[[
+            override this table, so that we don't have to
+            re-create a new table for this key, which would be costly.
+          --]]
+          cur_usage[cache_key].value = current_metric or 0
+          cur_usage[cache_key].expire_at = unix_time() + EXPIRATION[period]
+
+        else
+          cur_usage[0] = (cur_usage[0] or 0) + 1
+          cur_usage[cache_key] = {
```

Comment:
if we have two tables like `cur_usage_value` and `cur_usage_expire_at` then qw don't need to create small tables every time.